mozilla_django_oidc_db/backends.py

```diff
@@ -93,6 +69,18 @@ def __init__(self, *args, **kwargs) -> None:
         # AbstractUser.
         self.UserModel = cast(AbstractUser, get_user_model())
 
+    @property
+    def config(self):
+        if (configuration := getattr(self, "_config", None)) is None:
+            assert getattr(self, "request") is not None, (
+                "The request must be loaded from the authenticate entrypoint. It looks like "
+                "you're trying to access configuration before this was called."
+            )
+
+            configuration = lookup_config(self.request)
+            self._config = configuration
```

Comment:
can we not keep the behaviour that `authenticate` is responsible for resolving the config? I deliberately refactored the reliance on `self.request` out of the authentication backend, since the request itself should not be necessary, only the config instance is.

OTOH the base class also sets it... Either way, I don't think we need the tight coupling here to the request and having a setter for `self.config` that gets called during authenticate would be more architecturally sound.